src/spawn.cpp

```diff
@@ -268,22 +273,26 @@ void Spawn::checkSpawn()
 	}
 }
 
+void Spawn::scheduleSpawn(uint32_t spawnId, spawnBlock_t& sb, uint16_t interval)
+{
+	if (interval <= 0) {
+		spawnMonster(spawnId, sb.mType, sb.pos, sb.direction);
+	} else {
+		g_game.addMagicEffect(sb.pos, CONST_ME_TELEPORT);
+		g_scheduler.addEvent(createSchedulerTask(1400, std::bind(&Spawn::scheduleSpawn, this, spawnId, sb, interval - 1400)));
+	}
+}
+
 void Spawn::cleanup()
 {
 	auto it = spawnedMap.begin();
 	while (it != spawnedMap.end()) {
 		uint32_t spawnId = it->first;
 		Monster* monster = it->second;
 		if (monster->isRemoved()) {
-			if (spawnId != 0) {
-				spawnMap[spawnId].lastSpawn = OTSYS_TIME();
-			}
-
+			spawnMap[spawnId].lastSpawn = OTSYS_TIME();
```

Comment:
why did you remove that spawnId check?

Reply:
it was the "fix" for infinite respawn bug if monster walked away from the spawn, or was lured away

https://github.com/opentibiabr/otservbr-global/commit/34a20be159d84fd35877a49e2ca6dca4e9f519e3#diff-eecbbd78a76c67f252961d084417115a